Sources/SwiftParser/Declarations.swift

```diff
@@ -439,6 +464,36 @@ extension Parser {
   }
 }
 
+extension Parser {
+  mutating func parseUsingDeclaration(
+    attrs: DeclAttributes,
+    introducerHandle handle: RecoveryConsumptionHandle
+  ) -> RawUsingDeclSyntax {
+    precondition(attrs.attributes.isEmpty)
+    precondition(attrs.modifiers.isEmpty)
+
```

Comment:
I don’t think we necessarily need to crash the parser if `attrs.attributes` or `attrs.modifiers` is not empty. instead, we can shove them into `unexpectedBeforeUsingKeyword`. If for some reason they end up being non-nil that gives the user a better diagnostic that they can go off (plus doesn’t crash source tooling).

Reply:
@ahoppen does https://github.com/swiftlang/swift-syntax/pull/3087/files#diff-c777f9d3f9b258a7fc0c0f38f58e174c935a477c5bc8612a46c78bb23beba172R472-R493 makes sense or there is a better way?

Reply:
Looks good to me. Thank you.